Tests/Client/Unit Tests/WebSocket/WebSocketProviderMock.swift

```diff
@@ -0,0 +1,126 @@
+//
+//  WebSocketProviderMock.swift
+//  StreamChatClientTests
+//
+//  Created by Alexey Bukhtin on 01/05/2020.
+//  Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import XCTest
+@testable import StreamChatClient
+
+final class WebSocketProviderMock: WebSocketProvider {
+    let request: URLRequest
+    var isConnected: Bool = false
+    let callbackQueue: DispatchQueue
+    weak var delegate: WebSocketProviderDelegate?
+    private var connectionId: String?
+    
+    var failNextConnectCount = 0
+    let timeout = 3
```

Comment:
Your teammates would really appreciate if you put comments to these so we know how to use them in our tests :)